proposals/keptn.adoc

```diff
@@ -0,0 +1,306 @@
+=== Keptn CNCF Incubator Project Proposal
+
+*Name of Project:* keptn
+
+*Description:*
+
+Keptn is a control plane for continuous delivery and operations providing 
+a declarative approach towards day 1 to day n operations of cloud-native
+applications.
+
+Keptn is built based on the framework used internally at Dynatrace which allows
+us to run hundreds of production cluster without a dedicated operations team in
+a fully automated "NoOps" approach.
+
+Keptn solves the problem that traditional release pipelines are stitched together
+by point-to-point integrations between a number of tools. The average
+pipeline and operations automation toolchain consists of at least 7+ tools that
+are integrated. The current approach of building toolchains comes with a number
+of problems:
+
+* Integrations are hard to maintain as they are spread across the entire
+infrastructure
+* Onboarding a new service results in a lot of manual work for pipeline and
+operations automation definition
+* Most toolchains today make it hard to define enterprise-wide best practices
+for application delivery that can simply be applied across different
+applications.
+
+Keptn solves these problems by providing a configurable control plane to define
+day 1 to day n operations. The design is inspired by SDN stacks:
+
+* The application stack defines how the delivery and operations process should
+work using a shipyard file. Shipyard files define stages, release strategies,
+testing strategies, etc.
+* Keptn than provides a control plane that manages a GitOps repository for the
+stages defined in the shipyard file as well as managing Kubernetes namespaces
+and yaml files for services. Additionally, keptn uses a set of well-defined
+lifecycle events for each potential event that services can subscribe to. All
+events are defined as cloud events.
+* The integration layer used knative services to connect the Keptn control plane
+to all required tools to manage application delivery and operations. The
+configuration which tools to use is defined in a keptn uniform file.
+
+The approach of keptn has a number of advantages over traditional delivery pipelines and operation automation stacks:
+
+* The behavior of a delivery and automation workflow can be easily changed and
+is managed in a central place.
+* Using shipyard and uniform files the "how" is decoupled from the "what" which
```

Comment:
The idea of decoupling strategy from execution makes lots sense to me. 

Though from keptn's documentation, it's not clear how could user control or customize certain strategy. For instance, one user chose strategy of "canary-service" in shipyard.yaml, how could he specify the weight of canary increment step?

Reply:
In the `shipyard.yml` you could specify the canary weight increase and the time interval between increases as attributes to that strategy, like

```
strategy: "canary"
- increment: "10%"
- interval: "10m"
```